### PR TITLE
Delete source bin dir before building the snap (craft-414)

### PR DESCRIPTION
**Pull request by Blacksmoke16**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

Initial fix for https://bugs.launchpad.net/snapcraft/+bug/1936671.  Longer term something could be done to filter what files/directories should be copied from the host into the VM.  I.e. something like `.gitignore`.  But this at least resolves the immediate problem until if/when that happens.

### CODE REVIEW

File: snapcraft/plugins/v1/crystal.py [hunk 1]
```diff
@@ -92,6 +93,12 @@ def __init__(self, name, options, project):
             ]
         )
 
+    def pull(self):
+        super().pull()
+
+        with contextlib.suppress(FileNotFoundError):
+            shutil.rmtree(os.path.join(self.sourcedir, "bin"))
```
Comment (koffeinfrei) [inline snapcraft/plugins/v1/crystal.py:100]: Should we remove "lib" as well? To make sure the build isn't stale for all shards.

Comment (Blacksmoke16) [reply snapcraft/plugins/v1/crystal.py:100]: Yea probably.  Waiting to hear back from @sergiusens regarding https://bugs.launchpad.net/snapcraft/+bug/1936671/comments/8.  I.e. if this PR is obsolete.

Comment (sergiusens) [reply snapcraft/plugins/v1/crystal.py:100]: Sorry for the delay, I have replied in the bug

**Comment by Blacksmoke16** [comment]
Closing this as stale. Been almost 3 years.

**Pull request closed without merging**